### ISSUE TITLE
[Feature Request] SSO/OIDC

### ISSUE THREAD

**Issue by aevans1987** [andrii-kryvoviaz/slink#11]
OIDC/SSO Options for User Management for integration to things like Authentik?

**Comment by andrii-kryvoviaz** [comment]
I will have to look it up, since I still haven't integrated such things into my own homelab. Hence, my knowledge might be limited on how they are being integrated. But that sounds like a great feature to add.

**Comment by eddively** [comment]
I'd love this. Pocket ID is an easy(ish) one to get started.

**Comment by andrii-kryvoviaz** [comment]
> I'd love this. Pocket ID is an easy(ish) one to get started.

I'will take a look into this, thanks.

**Comment by Brramble** [comment]
@andrii-kryvoviaz would you be open to pull requests on this? I may start to work on this shortly.

**Comment by andrii-kryvoviaz** [comment]
@Brramble Sure, I will be glad to review and accept your pull request.

**Comment by andrii-kryvoviaz** [comment]
It took longer than expected 😅
At least now it is part of develop branch and is coming with the next minor release.

<img width="456" height="523" alt="Image" src="https://github.com/user-attachments/assets/32fe930a-7080-42b5-b9f7-ea03b02e15d4" />

This feature is configurable from the admin settings:

<img width="659" height="408" alt="Image" src="https://github.com/user-attachments/assets/4b717827-0882-4e11-990d-ac42705f8953" />

**Comment by jimmybrancaccio** [comment]
Looks awesome! Will there be an option to configure a generic OIDC provider? I use Zitadel and have Pocket ID and would love to use either (preferably Zitadel)!

Regardless, thanks for hard work on this! 🙌🏼

**Comment by andrii-kryvoviaz** [comment]
> Looks awesome! Will there be an option to configure a generic OIDC provider? I use Zitadel and have Pocket ID and would love to use either (preferably Zitadel)!
> 
> Regardless, thanks for hard work on this! 🙌🏼

Yeah, I thought on adding a generic provider, but decided to go with a predefined list instead. I’ll reconsider this approach before the actual release, since custom configuration (e.g., button color and icon) might be enough to provide the needed flexibility here.

**Comment by jimmybrancaccio** [comment]
> Yeah, I thought on adding a generic provider, but decided to go with a predefined list instead. I’ll reconsider this approach before the actual release, since custom configuration (e.g., button color and icon) might be enough to provide the needed flexibility here.

That would be super duper awesome to include a generic OIDC provider. I would think there's probably a generic OIDC icon or something that would work. 😁

Implementing the ability to input your own OIDC service information (callback URL, client ID and secret) via a generic OIDC provider setting would provide such a great flexibility for everyone not using those three predefined providers. 🙏🏼

**Comment by andrii-kryvoviaz** [comment]
> > Yeah, I thought on adding a generic provider, but decided to go with a predefined list instead. I’ll reconsider this approach before the actual release, since custom configuration (e.g., button color and icon) might be enough to provide the needed flexibility here.
> 
> That would be super duper awesome to include a generic OIDC provider. I would think there's probably a generic OIDC icon or something that would work. 😁
> 
> Implementing the ability to input your own OIDC service information (callback URL, client ID and secret) via a generic OIDC provider setting would provide such a great flexibility for everyone not using those three predefined providers. 🙏🏼

I decided to add support for generic provider configuration on top of predefined presets:

<img width="649" height="367" alt="Image" src="https://github.com/user-attachments/assets/3682af9d-c56a-4a4e-a08a-2fa61993be8e" />

<img width="373" height="251" alt="Image" src="https://github.com/user-attachments/assets/58e3c54e-1720-456b-976e-f5dd55a01fc9" />

Those will have no custom branding in comparison to the providers from preset list.

**Comment by jimmybrancaccio** [comment]
That's awesome, thank you so much for adding generic! 🙌🏼💛

**Comment by andrii-kryvoviaz** [comment]
This is already a part of `nightly` release (`latest` -> `nightly` if you want to test it). New features are frozen for the next release, only bug fixes, so pretty soon will be a new `v1.10.0`.  The only thing holds it back is that I started infrastructure [migration](https://github.com/andrii-kryvoviaz/slink/pull/166) and untill it's done and tested, I won't make the release.

**Comment by andrii-kryvoviaz** [comment]
Finally, it's a part of the new release, I may close it now 🎉
